src/AspNet.Security.OAuth.Paypal/PaypalAuthenticationHandler.cs

```diff
@@ -46,19 +51,14 @@ public PaypalAuthenticationHandler([NotNull] HttpClient client)
 
             var payload = JObject.Parse(await response.Content.ReadAsStringAsync());
 
-            identity.AddOptionalClaim(ClaimTypes.NameIdentifier, PaypalAuthenticationHelper.GetIdentifier(payload), Options.ClaimsIssuer)
-                    .AddOptionalClaim(ClaimTypes.Name, PaypalAuthenticationHelper.GetFullName(payload), Options.ClaimsIssuer)
-                    .AddOptionalClaim(ClaimTypes.GivenName, PaypalAuthenticationHelper.GetGivenName(payload), Options.ClaimsIssuer)
-                    .AddOptionalClaim(ClaimTypes.Surname, PaypalAuthenticationHelper.GetFamilyName(payload), Options.ClaimsIssuer)
-                    .AddOptionalClaim(ClaimTypes.Email, PaypalAuthenticationHelper.GetEmail(payload), Options.ClaimsIssuer);
-
             var principal = new ClaimsPrincipal(identity);
-            var ticket = new AuthenticationTicket(principal, properties, Options.AuthenticationScheme);
+            var context = new OAuthCreatingTicketContext(principal, properties, Context, Scheme, Options, Backchannel, tokens, payload);
+            context.RunClaimActions(payload);
 
-            var context = new OAuthCreatingTicketContext(ticket, Context, Options, Backchannel, tokens, payload);
-            await Options.Events.CreatingTicket(context);
+            identity.AddClaim(new Claim(ClaimTypes.NameIdentifier, payload.Value<string>("user_id")?.Split('/')?.LastOrDefault(), ClaimValueTypes.String, Options.ClaimsIssuer));
```

Comment:
FYI, you can use `ClaimActions.MapCustomJson` to add claims contained in nested JSON objects. E.g: https://github.com/aspnet/Security/blob/dev/src/Microsoft.AspNetCore.Authentication.MicrosoftAccount/MicrosoftAccountOptions.cs#L32

Any chance we could use it everywhere we can?